scripts/linters/general_purpose_linter_test.py

```diff
@@ -143,615 +143,558 @@ class HTMLLintTests(test_utils.LinterTestBase):
-            ).perform_all_lint_checks()
+        linter = general_purpose_linter.GeneralPurposeLinter(
+            [INVALID_PARENT_HTML_FILEPATH], FILE_CACHE)
+        lint_task_report = linter.check_bad_patterns()
         self.assert_same_list_elements([
             'Line 13: Please do not access parent properties using '
             '$parent. Use the scope object for this purpose.'
-            ], self.linter_stdout)
-        self.assert_failed_messages_count(self.linter_stdout, 1)
+            ], lint_task_report.messages)
+        self.assert_failed_messages_count(lint_task_report.all_messages, 1)
 
 
 class JsTsLintTests(test_utils.LinterTestBase):
     """Test the JsTs lint functions."""
 
     def test_invalid_use_of_browser_explore(self):
-        with self.print_swap:
-            general_purpose_linter.GeneralPurposeLinter(
-                [INVALID_EXPLORE_FILEPATH], FILE_CACHE, True
-            ).perform_all_lint_checks()
+        linter = general_purpose_linter.GeneralPurposeLinter(
+            [INVALID_EXPLORE_FILEPATH], FILE_CACHE)
+        lint_task_report = linter.check_bad_patterns()
         self.assert_same_list_elements(
             ['Line 30: In tests, please do not use browser.explore().'],
-            self.linter_stdout)
-        self.assert_failed_messages_count(self.linter_stdout, 1)
+            lint_task_report.messages)
+        self.assert_failed_messages_count(lint_task_report.all_messages, 1)
 
     def test_invalid_use_of_browser_pause(self):
-        with self.print_swap:
-            general_purpose_linter.GeneralPurposeLinter(
-                [INVALID_PAUSE_FILEPATH], FILE_CACHE, True
-            ).perform_all_lint_checks()
+        linter = general_purpose_linter.GeneralPurposeLinter(
+            [INVALID_PAUSE_FILEPATH], FILE_CACHE)
+        lint_task_report = linter.check_bad_patterns()
         self.assert_same_list_elements(
             ['Line 30: In tests, please do not use browser.pause().'],
-            self.linter_stdout)
-        self.assert_failed_messages_count(self.linter_stdout, 1)
+            lint_task_report.messages)
+        self.assert_failed_messages_count(lint_task_report.all_messages, 1)
 
     def test_invalid_use_of_browser_sleep(self):
-        with self.print_swap:
-            general_purpose_linter.GeneralPurposeLinter(
-                [INVALID_SLEEP_FILEPATH], FILE_CACHE, True
-            ).perform_all_lint_checks()
+        linter = general_purpose_linter.GeneralPurposeLinter(
+            [INVALID_SLEEP_FILEPATH], FILE_CACHE)
+        lint_task_report = linter.check_bad_patterns()
         self.assert_same_list_elements(
             ['Line 30: In tests, please do not use browser.sleep().'],
-            self.linter_stdout)
-        self.assert_failed_messages_count(self.linter_stdout, 1)
+            lint_task_report.messages)
+        self.assert_failed_messages_count(lint_task_report.all_messages, 1)
```

Comment:
As mentioned above, please remove these `assert_same_list_elements` & `assert_failed_messages_count` and assert the complete message and other values of the taskResult*